src/xpk/core/kjob.py

```diff
@@ -0,0 +1,186 @@
+  kind: JobTemplate
+  metadata:
+    name: {name}
+    namespace: default
+  template:
+    spec:
+      parallelism: {parallelism}
+      completions: {completions}
+      completionMode: Indexed
+      template:
+        spec:
+          containers:
+            - name: {container}
+              image: {image}
+          restartPolicy: OnFailure"""
+
+app_profile_yaml = """
+apiVersion: kjobctl.x-k8s.io/v1alpha1
+kind: ApplicationProfile
+metadata:
+  name: {name}
+  namespace: default
+spec:
+  supportedModes:
+    - name: Slurm
+      template: {template}
+      requiredFlags: []
+"""
+
+
+def verify_kjob_installed(args: Namespace) -> None:
+  """Check if kjob is installed. If not provide user with proper communicate and exit.
+  Args:
+    args - user provided arguments.
+  Returns:
+    None
+  """
+  command = "kubectl-kjob help"
+  task = "Verify kjob installation "
+  verify_kjob_installed_code, _ = run_command_for_value(command, task, args)
+
+  if verify_kjob_installed_code == 0:
+    xpk_print("kjob found")
+
+  if verify_kjob_installed_code != 0:
+    xpk_print(
+        " kjob not found. Please follow"
+        " https://github.com/kubernetes-sigs/kueue/blob/main/cmd/experimental/kjobctl/docs/installation.md"
+        " to install kjob."
+    )
+    xpk_exit(verify_kjob_installed_code)
+
+
+def create_app_profile_instance(args: Namespace) -> None:
+  """Create new AppProfile instance on cluster with default settings.
+
+  Args:
+    args - user provided arguments
+  Returns:
+    None
+  """
+  yml_string = app_profile_yaml.format(
+      name=APP_PROFILE_TEMPLATE_DEFAULT_NAME,
+      template=JOB_TEMPLATE_DEFAULT_NAME,
+  )
+
+  tmp = write_tmp_file(yml_string)
+  command = f"kubectl apply -f {str(tmp.file.name)}"
+  return_code = run_command_with_updates(command, "Creating AppProfile", args)
+  if return_code != 0:
+    xpk_exit(return_code)
+
+
+def create_job_template_instance(args: Namespace) -> None:
+  """Create new JobTemplate instance on cluster with default settings.
+
+  Args:
+    args - user provided arguments
+  Returns:
+    None
+  """
+  yml_string = job_template_yaml.format(
+      name=JOB_TEMPLATE_DEFAULT_NAME,
+      parallelism=JOB_TEMPLATE_DEFAULT_PARALLELISM,
+      completions=JOB_TEMPLATE_DEFAULT_COMPLETIONS,
+      container=JOB_TEMPLATE_DEFAULT_CONT_NAME,
+      image=JOB_TEMPLATE_DEFAULT_IMG,
+  )
+
+  tmp = write_tmp_file(yml_string)
+  command = f"kubectl apply -f {str(tmp.file.name)}"
+  return_code = run_command_with_updates(command, "Creating JobTemplate", args)
+  if return_code != 0:
+    xpk_exit(return_code)
+
+
+def apply_kjob_crds(args: Namespace) -> None:
+  """Apply kjob CRDs on cluster.
+
+  This function downloads kjob CRDs files from kjob repo,
+  builds them with kustomize and then applies result on cluster.
+  It creates all neccessary kjob CRDs.
+
+  Args:
+    args - user provided arguments
+  Returns:
+    None
+  """
+  kjob_kustomize_path = tempfile.mkdtemp()
```

Comment:
Isn't it better to download files one by one and apply. them one by one?
If there is any issue with any files we can report it better?

Anyways,I beleive this method can be splitted in smaller pieces with a descriptive names. Example:
"download_crds"
"download_kustomization"// or something better
"apply_crds"

Reply:
I think if downloading any of the files fail, we should faill whole installation.
Same with applying. What I did here, is I repeated make install commend from kjob: https://github.com/kubernetes-sigs/kueue/blob/5203ac9eb69226a1ae3b6ba6d361eaed7ef3dc8c/cmd/experimental/kjobctl/Makefile#L160C1-L162C69 I did not want  to copy whole repo here or Makefile and invoke make from our code, that is why kustomization is used.